### DISCUSSION TITLE
Big Query Impersonate User to have Access to DataSource

### DISCUSSION THREAD

**Discussion by codegeekjk** [apache/superset#18269] [Q&A / Help]
Hello All,

We have big query as one of the datasource in superset, currently access to dataset is based on the google-group.
From superset we want to impersonate user so only the authorised users should be able to access the datasets.

Currently the user impersonate is available for  Hive and Presto, how can I do that for BigQuery ?
Anywork arounds avaliable so we can get this working ?

**Comment by cseHdz** [comment]
This would be quite helpful for us as well. 

We have dataset and column level permissions well-defined in Google BigQuery - Without pass through, we would be looking at many BigQuery connections with custom service accounts for each "group" of datasets/columns.

This would take care of the issue in  #9182 as permissions in IAM would be honored in Supereset.

**Comment by lihan** [comment]
+1 for this

**Comment by schickler-t-ziegler** [comment]
+1 for this

**Comment by lihan** [comment]
@zuzana-vej 

Be able to use an end user's Google account to access BigQuery enables BigQuery Audit support and row-level security.

Perhaps we can store  Google Cloud access token on the user table and use that to access the Bigquery API?

Related:
* https://cloud.google.com/docs/authentication/end-user#authenticating_and_calling_the_api
* https://cloud.google.com/sdk/gcloud/reference/auth/application-default/print-access-token

**Comment by adhi-moves** [comment]
+1

**Comment by lihan** [comment]
This would be still a huge benefit for teams where they have customers data in it. Any soft of directions on implementing would be great.

**Comment by edbizarro** [comment]
+1

**Comment by Agos88** [comment]
+1

**Comment by rogerzs** [comment]
hey @codegeekjk , are you still trying implement this auth feature?
